source/agora/network/Manager.d

```diff
@@ -1063,6 +1066,39 @@ public class NetworkManager
         return headers_updated;
     }
 
+    /// Retrieve any missing enrollments from randomly selected nodes
+    public void getEnrollments (EnrollmentManager enroll_man,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        // TODO: This should be part of the configuration.
+        static uint MinNodes = 3;
```

Comment:
I thought that getting enrollments from all other nodes could impose a burden on the network. So I sampling the nodes which will get the request, or `getEnrollments`. @Geod24  How do you think of this approach?

Reply:
Getting all enrollments also means that we most likely always have needless data transfer even when we're up to date, which isn't optimal.

Reply:
OK, I see what you mean. I will find a way to store missing enrollments that will be requested later from other nodes.

Reply:
Had a different approach and solved it.